src/triage/component/catwalk/storage.py

```diff
@@ -213,123 +243,179 @@ def matrix_with_sorted_columns(self, columns):
-
-    def _write_hdf_to_buffer(self, df):
-        with pandas.HDFStore(
-                "data.h5",
-                mode="w",
-                driver="H5FD_CORE",
-                driver_core_backing_store=0) as out:
-            out["matrix"] = df
-            return out._handle.get_file_image()
+        path_parsed = urlparse(self.matrix_path)
+        scheme = path_parsed.scheme  # If '' of 'file' is a regular file or 's3'
+
+        if scheme in ('', 'file'):  # Local file
+            matrix = pd.read_hdf(self.matrix_path)
+        else:
+            raise ValueError(f"""
+                  URL scheme not supported:
+                  {scheme} (from {self.matrix_path})
+            """)
+        # Is the index already in place?
+        if matrix.index.name not in self.metadata['indices']:
+            matrix.set_index(self.metadata['indices'], inplace=True)
+
+        return matrix
 
     def save(self, project_path, name):
-        with smart_open.smart_open(os.path.join(project_path, name + ".h5"), "wb") as f:
-            f.write(self._write_hdf_to_buffer(self.matrix))
-        self.save_yaml(self.metadata, project_path, name)
+        path_parsed = urlparse(project_path)
+        scheme = path_parsed.scheme  # If '' of 'file' is a regular file or 's3'
+
+        if self.scheme in ('', 'file'):  # Local file
+            with open(os.path.join(project_path, name + ".h5"), "w") as f:
+                self.matrix.to_hdf(f, format='table', mode='w')
+        else:
+            raise ValueError(f"""
+                  URL scheme not supported:
+                  {scheme} (from {os.path.join(project_path, name + '.h5')})
+            """)
+
+        self.save_metadata(self.metadata, project_path, name)
 
 
 class CSVMatrixStore(MatrixStore):
+
+    def __init__(self, matrix_path=None, metadata_path=None):
+        super().__init__(matrix_path, metadata_path)
+
     def _get_head_of_matrix(self):
+        path_parsed = urlparse(self.matrix_path)
+        scheme = path_parsed.scheme  # If '' of 'file' is a regular file or 's3'
+
         try:
-            head_of_matrix = pandas.read_csv(self.matrix_path, nrows=1)
+            if scheme in ('','file'):  # Local file
+                with open(self.matrix_path, "r") as f:
+                    head_of_matrix = pd.read_csv(f, nrows=1)
+            elif scheme == 's3':
+                s3 = s3fs.S3FileSystem()
+                with s3.open(self.matrix_path, 'rb') as f:
+                    head_of_matrix = pd.read_csv(f, nrows=1)
+            else:
+                raise ValueError(f"URL scheme not supported: {scheme} (from {self.matrix_path})")
+
             head_of_matrix.set_index(self.metadata['indices'], inplace=True)
-            self._head_of_matrix = head_of_matrix
-        except pandas.error.EmptyDataError:
-            self._head_of_matrix = None
+
+        except FileNotFoundError as fnfe:
+            logging.error(f"Matrix isn't there: {fnfe}")
```

Comment:
unless you really want to suppress the traceback, rather than capture the exception instance, you can simply use `logging.exception` –

    except FileNotFoundError:
        logging.exception("Matrix isn't there")
        …

Reply:
I didn't knew this